pandas/core/internals/managers.py

```diff
@@ -1185,6 +1185,17 @@ def _iset_single(
         self.blocks = new_blocks
         return
 
+    def column_setitem(self, loc: int, idx: int | slice | np.ndarray, value) -> None:
+        """
+        Set values ("setitem") into a single column (not setting the full column).
+
+        This is a method on the BlockManager level, to avoid creating an
+        intermediate Series at the DataFrame level (`s = df[loc]; s[idx] = value`)
+        """
+        col_mgr = self.iget(loc)
+        new_mgr = col_mgr.setitem((idx,), value)
+        self.iset(loc, new_mgr._block.values, inplace=True)
```

Comment:
i think you can use _setitem_single here?

Reply:
`iset` will indeed call `_iset_single` in this case, but it still does some useful preprocessing that we might not want to repeat here I think (eg converting `loc` in a `blkno`)